include/fiction/technology/sidb_dynamic_gate_library.hpp

```diff
@@ -0,0 +1,1244 @@
+//
+// Created by Jan Drewniok 20.09.23.
+//
+
+#ifndef FICTION_SIDB_DYNAMIC_GATE_LIBRARY_HPP
+#define FICTION_SIDB_DYNAMIC_GATE_LIBRARY_HPP
+
+#include "fiction/algorithms/path_finding/distance.hpp"
+#include "fiction/algorithms/physical_design/design_sidb_gates.hpp"
+#include "fiction/algorithms/physical_design/is_gate_design_impossible.hpp"
+#include "fiction/algorithms/simulation/sidb/is_operational.hpp"
+#include "fiction/io/read_sqd_layout.hpp"
+#include "fiction/technology/cell_technologies.hpp"
+#include "fiction/technology/fcn_gate_library.hpp"
+#include "fiction/technology/sidb_surface.hpp"
+#include "fiction/technology/sidb_surface_analysis.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/utils/layout_utils.hpp"
+
+#include <array>
+#include <stdexcept>
+
+namespace fiction
+{
+/**
+ * This struct stores the parameter to design gates on the fly.
```

Comment:
Please be consistent about whether you are designing gates "on the fly" or "dynamically".